src/abTestingInfra/Experiment/index.tsx

```diff
@@ -0,0 +1,133 @@
+import React from "react";
+import { assertNameIsValid } from "../utils";
+import { ABTestingContextConsumer } from "../ABTestingContext";
+import PropTypes from "prop-types";
+import { recordPlay } from "../emitter";
+import { ABTestingPayload, VariantProps, Variants } from "../types";
+import ExperimentContext from "../ExperimentContext";
+
+class Experiment extends React.Component {
```

Comment:
Why are we using a class component vs functional?

Reply:
@anulman I need you help with this, can we do a quick pair programming session?